src/astar/astar_driver.cpp

```diff
@@ -117,6 +117,12 @@ void do_pgr_astarManyToMany(
 
         graphType gType = directed? DIRECTED: UNDIRECTED;
 
+        // sorting the edges in an ascending order of their id, before creating the graph
+        std::sort(edges, edges + total_edges,
```

Comment:
So, I see your solution is within the drivers
All say: "before creating the graph"
So I ask, what if instead of putting the same `sort`  over and over and over everywhere, you put it inside the graph creation function.
Note that because some of the functions do not need the sorting (sorting takes NlogN time)then the graph creation function needs an additional parameter basically a bool that indicates that the info should be sorted
You can work it out as follows:
- Find the code where the graph gets created
- Add the additional parameter as [default argument](https://en.cppreference.com/w/cpp/language/default_arguments) (should be the last one of the parameters)
- Set the default value to false. that way you cover:
  - you don't need to edit drivers where the sorting is not needed
  - modifying a driver would be a matter to add `,true` 
  - work this one out together with astar_driver.cpp
- once it works make unique commit when adding that general sort